File: src/types/note.ts
export interface Note {
  id: string;
  title: string;
  path: string;
  categoryId: string;
  createdAt: string;
  updatedAt: string;
  tags: string[];
}

export interface NoteContent {
  id: string;
  title: string;
  content: string;
  frontMatter: FrontMatter;
}

export interface FrontMatter {
  title?: string;
  tags: string[];
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateNoteRequest {
  categoryId: string;
  title: string;
}

export interface SaveNoteRequest {
  path: string;
  content: string;
}

// ================== Tree Node Types ==================

export type TreeNodeType = 'folder' | 'file';

export interface TreeNode {
  id: string;
  name: string;
  path: string;
  type: TreeNodeType;
  depth: number;
  createdAt: string;
  updatedAt: string;
  tags?: string[];
  isEmpty?: boolean;
}

export interface CreateFolderRequest {
  parentPath: string;
  name: string;
}

export interface CreateNoteInFolderRequest {
  parentPath: string;
  title: string;
}

File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Note {
    pub id: String,
    pub title: String,
    pub path: String,
    pub category_id: String,
    pub created_at: String,
    pub updated_at: String,
    pub tags: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NoteContent {
    pub id: String,
    pub title: String,
    pub content: String,
    pub front_matter: FrontMatter,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct FrontMatter {
    pub title: Option<String>,
    pub tags: Vec<String>,
    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct CreateNoteRequest {
    pub category_id: String,
    pub title: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct SaveNoteRequest {
    pub path: String,
    pub content: String,
}